src/pytorch/messages/passMessages.py

```diff
@@ -0,0 +1,60 @@
+from torch import ops, Tensor
+
+
+def passMessages(neighbors: Tensor, messages: Tensor, states: Tensor) -> Tensor:
+    '''
+    Pass messages between the neighbor atoms.
+
+    Given a set of `num_atoms` atoms (each atom has a state with `num_features`
+    features) and a set of `num_neighbors` neighbor atom pairs (each pair has a
+    message with `num_features` features), the messages of the pairs are added
+    to the corresponding atom states.
+
+    Parameters
+    ----------
+    neighbors: `torch.Tensor`
+        Atom pair indices. The shape of the tensor is `(2, num_pairs)`.
+        The indices can be `[0, num_atoms-1)` or `-1` (ignored pairs).
+        See for the documentation of `NNPOps.neighbors.getNeighborPairs` for
+        details.
+    messages: `torch.Tensor`
+        Atom pair messages. The shape of the tensor is `(num_pairs, num_features)`.
+        For efficient, `num_features` has to be a multiple of 32 and <= 1024.
```

Comment:
Are those limitations really necessary?  It's very common for the number of features not to be a multiple of 32.

Reply:
In contrary, the number of internal features is always factor of 32 (e.g. in TorchMD-NET, I have seen usage of 64, 128, 256). GPU computes in warps of 32 threads, so it is the best to match that patter for computational efficiency.

Reply:
I frequently create models that don't satisfy those requirements, including in TorchMD-Net.  For example, I've trained models with 48 or 80 features per layer.

Reply:
Why?

Reply:
I don't understand your question.  Why not?  The number of features is a hyperparameter.  It's one of many hyperarameters you tune to balance training accuracy, overfitting, speed, and memory use.  Why place arbitrary limits on it when there's no need to?